client/main.lua

```diff
@@ -1,274 +1,223 @@
-                    shouldClose = true,
-                },
-            }
-    else
-        SetupSubItems()
-        FinalMenuItems = deepcopy(Config.MenuItems)
-        for _, v in pairs(DynamicMenuItems) do
-            FinalMenuItems[#FinalMenuItems+1] = v
-        end
-
-    end
+local function IsPolice()
+    return (PlayerData.job.type == "leo" and PlayerData.job.onduty)
 end
 
-local function setRadialState(bool, sendMessage, delay)
-    -- Menuitems have to be added only once
+local function IsEMS()
+    return (PlayerData.job.type == "medic" and PlayerData.job.onduty)
+end
 
-    if bool then
-        TriggerEvent('qb-radialmenu:client:onRadialmenuOpen')
-        SetupRadialMenu()
+-- Events
+RegisterNetEvent('radialmenu:client:deadradial', function(isDead)
+    if isDead then
+        local ispolice, isems = IsPolice(), IsEMS()
+        if not ispolice or isems then return lib.disableRadial(true) end
+        lib.clearRadialItems()
+        lib.addRadialItem({
+            id = 'emergencybutton2',
+            label = Lang:t("options.emergency_button"),
+            icon = 'circle-exclamation',
+            onSelect = function (_, _)
+                if ispolice then
+                    TriggerEvent('police:client:SendPoliceEmergencyAlert')
+                elseif isems then
+                    TriggerEvent('ambulance:client:SendAmbulanceEmergencyAlert')
+                end
+                lib.hideRadial()
+            end
+        })
     else
-        TriggerEvent('qb-radialmenu:client:onRadialmenuClose')
+        lib.clearRadialItems()
+        SetupRadialMenu()
+        lib.disableRadial(false)
     end
+end)
 
-    SetNuiFocus(bool, bool)
-    if sendMessage then
-        SendNUIMessage({
-            action = "ui",
-            radial = bool,
-            items = FinalMenuItems
-        })
+RegisterNetEvent('radialmenu:client:ChangeSeat', function(id, label)
+    local Veh = GetVehiclePedIsIn(cache.ped, false)
+    local IsSeatFree = IsVehicleSeatFree(Veh, id - 2)
+    local speed = GetEntitySpeed(Veh)
+    local HasHarness = exports['qb-smallresources']:HasHarness()
+    if HasHarness then
+        return QBCore.Functions.Notify(Lang:t("error.race_harness_on"), 'error')
     end
-    if delay then Wait(500) end
-    inRadialMenu = bool
-end
-
--- Command
 
-RegisterCommand('radialmenu', function()
-    if ((IsDowned() and IsPoliceOrEMS()) or not IsDowned()) and not PlayerData.metadata["ishandcuffed"] and not IsPauseMenuActive() and not inRadialMenu then
-        setRadialState(true, true)
-        SetCursorLocation(0.5, 0.5)
+    if not IsSeatFree then
+        QBCore.Functions.Notify(Lang:t("error.seat_occupied"), 'error')
     end
+    
+    local kmh = speed * 3.6
+    
+    if kmh > 100.0 then
+        QBCore.Functions.Notify(Lang:t("error.vehicle_driving_fast"), 'error')
+    end
+    
+    SetPedIntoVehicle(cache.ped, Veh, id - 2)
+    QBCore.Functions.Notify(Lang:t("info.switched_seats", {seat = label}))
```

Comment:
```suggestion
    if not IsSeatFree then
        QBCore.Functions.Notify(Lang:t("error.seat_occupied"), 'error')
        return
    end
    
    local kmh = speed * 3.6
    
    if kmh > 100.0 then
        QBCore.Functions.Notify(Lang:t("error.vehicle_driving_fast"), 'error')
        return
    end
    
    SetPedIntoVehicle(cache.ped, Veh, id - 2)
    QBCore.Functions.Notify(Lang:t("info.switched_seats", {seat = label}))
```

Guard clauses require a return statement or the code underneath will execute (which defeats the purpose of having them).

Reply:
Oh fuck i forgor them mb